portal/views/portal.py

```diff
@@ -489,21 +491,36 @@ def challenge_identity(user_id=None, next_url=None, merging_accounts=False):
 @portal.route('/initial-queries', methods=['GET','POST'])
 def initial_queries():
     """Initial consent terms, initial queries view function"""
-    if request.method == 'POST':
-        # data submission all handled via ajax calls from initial_queries
-        # template.  assume POST can only be sent when valid.
-        current_app.logger.debug("POST initial_queries -> next_after_login")
-        return next_after_login()
-
     user = current_user()
     if not user:
         # Shouldn't happen, unless user came in on a bookmark
         current_app.logger.debug("initial_queries (no user!) -> landing")
         return redirect(url_for('portal.landing'))
     if user.deleted:
         abort(400, "deleted user - operation not permitted")
+    if request.method == 'POST':
+        """
+         data submission all handled via ajax calls from initial_queries
+         template.  assume POST can only be sent when valid.
+        """
+        current_app.logger.debug("POST initial_queries -> next_after_login")
+        return next_after_login()
+    elif len(Coredata().still_needed(user)) == 0:
+        # also handle the situations that resulted from: 1. user refreshing the browser or
+        # 2. exiting browser and resuming session thereafter
+        # In both cases, the request method is GET, 
+        # hence a redirect back to initial-queries page won't ever reach the above check 
+        # specifically for next_after_login based on the request method of POST
+        # NB: user needs to have provided consent before we can check for next_after_login
+        # case in point:
+        # user has pre-selected clinic via access code BUT has not provided consent yet
```

Comment:
I get the need for the `elif` - but not the hand selected lookup for `UserConsent`.  There may well be a roll or product where that doesn't make sense.

If `len(Coredata().still_needed(user)) == 0` - you can safely assume we have everything needed, right?